include/nrm-tools.h

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+ * Copyright 2019 UChicago Argonne, LLC.
+ * (c.f. AUTHORS, LICENSE)
+ *
+ * This file is part of the libnrm project.
+ * For more info, see https://github.com/anlsys/libnrm
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *******************************************************************************/
+
+#include "nrm.h"
+
+#ifndef NRM_TOOLS_H
+#define NRM_TOOLS_H 1
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*******************************************************************************
+ * Command Line Parsing Helpers
+ ******************************************************************************/
+
+#define NRM_TOOLS_ARGS_FLAG_FREQ 1
+#define NRM_TOOLS_ARGS_FLAG_MAX 2
+
+typedef struct nrm_tools_args_s {
+	/* needs to be set ahead of time */
+	char *progname;
+	int flags;
+	/* common options */
+	nrm_string_t upstream_uri;
+	unsigned int pub_port;
+	unsigned int rpc_port;
+	int log_level;
+	int ask_help;
+	int ask_version;
+	/* beginning of extra options */
+	double freq;
+} nrm_tools_args_t;
+
+/* parse command-line arguments, consuming them,
+ * at the end of the function, argc and argv point to unparsed options or
+ * positional arguments.
```

Comment:
I don't really understand this comment. `argc` is passed by value so it won't change? :confused: